Content.Server/Administration/WatchlistWebhookSystem.cs

```diff
@@ -0,0 +1,142 @@
+using Content.Server.Administration.Notes;
+using Content.Server.Database;
+using Content.Server.Discord;
+using Content.Shared.CCVar;
+using Robust.Server;
+using Robust.Server.Player;
+using Robust.Shared.Configuration;
+using Robust.Shared.Network;
+using System.Linq;
+using Timer = Robust.Shared.Timing.Timer;
+
+namespace Content.Server.Administration;
+
+/// <summary>
+///     This system sends a Discord webhook notification whenever a player with an active
+///     watchlist joins the server.
+/// </summary>
+public sealed class WatchlistWebhookSystem : EntitySystem
+{
+    [Dependency] private readonly IAdminNotesManager _adminNotes = default!;
+    [Dependency] private readonly IBaseServer _baseServer = default!;
+    [Dependency] private readonly IConfigurationManager _cfg = default!;
+    [Dependency] private readonly DiscordWebhook _discord = default!;
+    [Dependency] private readonly INetManager _netManager = default!;
+    [Dependency] private readonly IPlayerManager _playerManager = default!;
+
+    private WebhookIdentifier? _webhookIdentifier;
+    private List<WatchlistConnection> watchlistConnections = new();
+    // true when a timer is running for the currently buffered connections, and another should not be started
+    private bool buffering = false;
+
+    public override void Initialize()
+    {
+        Subs.CVar(_cfg, CCVars.DiscordWatchlistConnectionWebhook, SetWebhookIdentifier, true);
+        _netManager.Connected += OnConnected;
+        base.Initialize();
+    }
+
+    private void SetWebhookIdentifier(string? value)
+    {
+        if (!string.IsNullOrWhiteSpace(value))
+            _discord.GetWebhook(value, data => _webhookIdentifier = data.ToIdentifier());
+    }
+
+    private async void OnConnected(object? sender, NetChannelArgs e)
+    {
+        var watchlists = await _adminNotes.GetActiveWatchlists(e.Channel.UserId);
+
+        if (watchlists.Count == 0)
+            return;
+
+        var session = _playerManager.GetSessionByChannel(e.Channel);
```

Comment:
Since you need the `IPlayerSession` here, it's better if you listen for `IPlayerManager.PlayerStatusChanged` instead.